link: allow direct aliasing

yoshuawuyts opened the pull request:
Further iteration of #20.

This allows for direct aliasing of `man-n --link`, removing the extra overhead of running `eval` every time a new shell is started. It also exposes a familiar interface to new users by using `alias`. It does add some recursion in the code, but I think it's transparent what's going on as long as users are familiar with `shift(1)`.

This also moves all CLI flags to the `case` statement, making the user interface more transparent and eliminating the need for the undocumented `--linked` flag.

The newly introduced `switch` routine can be replaced with something smarter in the future, possibly allowing to check local deps / remove the need to use `n` in commands. This patch introduces a location to start that work.

Thanks!

## Changes
- __link__: `man-n --link` can now directly be aliased
- __link__: add `-l` as shorthand
- __docs__: update alias example in readme
- __help__: update `--link` usage examples
- __link__: fall back to npm search if `man(1)` yields no results

jbnicolai commented on bin/man-n:
```diff
@@ -22,38 +22,45 @@ cat << EOF
+  esac
+}
+
+switch "$@"
```
Is there a reason you prefer to put it in a function?

yoshuawuyts replied:
We're calling this recursively on [line 43](https://github.com/man-n/man-n/pull/21/files#diff-a942ee1af8cafadc0e5b2bc79af14388R43).

jbnicolai replied:
Ah I completely missed that. Nice one!

jbnicolai replied:
Small note for simplicity's sake: rather than calling `switch` at this point, you could also simply call the original `$0` with `$@` at that point. This would move the recursion to the top of the script, making it perhaps simpler to reason about.

jbnicolai commented on bin/man-n:
```diff
@@ -22,38 +22,45 @@ cat << EOF
+  shift
+  if [ "$1" = "n" ]; then
+    shift
+    switch "$@"
```
If at the top you'd place

```
readonly command="$0"
```

Couldn't this become

``` bash
shift
"$command" "$@"
```

?

yoshuawuyts commented:
@jbnicolai updated with your feedback! Code should be a bit easier to reason about now.

yoshuawuyts commented:
Code from https://github.com/man-n/man-n/pull/23 was added.

## Changes
- **link**: fall back to npm search if man(1) yields no results

wooorm commented:
I really like this but I’m not sure these commands should be in the usage section? I don’t really thing people will try to `man-n --link ls` instead of `man ls`. Isn't it more of an implementation detail?

Also, redirecting both stdout and stderr to /dev/null is pretty dirty/simple. Also: falling back to npm on **any** man page failure feels like that too.

I suggest basing the `$original` or `$manN` switch on `man -w`, which outputs the following if a man page exists locally:

``` bash
🚀  ~: man -w mdast
# /Users/wooorm/.npm-packages/share/man/man1/mdast.1

🚀  ~: man -w browserify
# No manual entry for browserify
```

Other than these two: :+1:

yoshuawuyts commented:
Fixed an error with pipe redirection (`stdin` was accidentally also piped to `/dev/null`) and removed inline `--link` examples. @wooorm I'm not quite sure what you  mean with `man -w`; would you be OK if this patched got merged and you'd apply the `-w` changes on top? 

Also: I'll be gone for the next few days, you know: changing continents. This is my last contribution for now.

wooorm commented:
Sure, I’ll be able to merge and patch later today! Enjoy the continent change 😉

yoshuawuyts commented:
:tada: thanks!

wooorm commented:
Closed by 425c72a.
